Client/Frontend/Browser/ImageCache/WebImageCache.swift

```diff
@@ -0,0 +1,122 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import SDWebImage
+
+final class WebImageCache: ImageCacheProtocol {
+    
+    private let webImageManager: SDWebImageManager
+    
+    private let privacyProtection: PrivacyProtectionProtocol
+    
+    typealias ReturnAssociatedType = SDWebImageOperation
+    
+    deinit {
+        clearMemoryCache()
+        clearDiskCache()
+    }
+    
+    init(withPrivacyProtection privacyProtection: PrivacyProtectionProtocol, sandbox: String? = nil) {
+        self.privacyProtection = privacyProtection
+        
+        var imageCache: SDImageCache
+        
+        if let sandbox = sandbox {
+            imageCache = SDImageCache(namespace: sandbox, diskCacheDirectory: sandbox)
+        } else {
+            imageCache = SDImageCache.shared()
+        }
+        
+        let imageDownloader = SDWebImageDownloader.shared()
+        
+        webImageManager = SDWebImageManager(cache: imageCache, downloader: imageDownloader)
+    }
+    
+    @discardableResult func load(from url: URL, options: ImageCacheOptions = [], progress: ImageCacheProgress = nil, completion: ImageCacheCompletion = nil) -> SDWebImageOperation? {
+        let progressBlock: SDWebImageDownloaderProgressBlock = { (receivedSize, expectedSize, url) -> Void in
+            guard let url = url else {
+                return
+            }
+            
+            progress?(receivedSize, expectedSize, url)
+        }
+        
+        var webImageOptions = self.webImageOptions
+        
+        if options.contains(.lowPriority) {
+            webImageOptions.append(.lowPriority)
+        }
+        
+        let imageOperation = webImageManager.loadImage(with: url, options: SDWebImageOptions(webImageOptions), progress: progressBlock) { (image, data, error, webImageCacheType, _, imageURL) in
+            if let image = image, !self.privacyProtection.nonPersistent {
+                self.webImageManager.saveImage(toCache: image, for: url)
+            }
+            
+            let cacheType = self.mapImageCacheType(from: webImageCacheType)
+            completion?(image, data, error, cacheType, url)
+        }
+        
+        return imageOperation
+    }
+    
+    func isCached(_ url: URL) -> Bool {
+        if webImageManager.cacheKey(for: url) != nil {
```

Comment:
`return webImageManager.cacheKey(for: url) != nil`

Reply:
👍